packages/0x.js/src/order_watcher/remaining_fillable_calculator.ts

```diff
@@ -0,0 +1,71 @@
+    private _remainingMakerFeeAmount: BigNumber;
+    constructor(signedOrder: SignedOrder,
+                isMakerTokenZRX: boolean,
+                transferrableMakerTokenAmount: BigNumber,
+                transferrableMakerFeeTokenAmount: BigNumber,
+                remainingMakerTokenAmount: BigNumber) {
+        this._signedOrder = signedOrder;
+        this._isMakerTokenZRX = isMakerTokenZRX;
+        this._transferrableMakerTokenAmount = transferrableMakerTokenAmount;
+        this._transferrableMakerFeeTokenAmount = transferrableMakerFeeTokenAmount;
+        this._remainingMakerTokenAmount = remainingMakerTokenAmount;
+        this._remainingMakerFeeAmount = remainingMakerTokenAmount.times(signedOrder.makerFee)
+                                                                 .dividedToIntegerBy(signedOrder.makerTokenAmount);
+    }
+    public computeRemainingMakerFillable(): BigNumber {
+        if (this.hasSufficientFundsForFeeAndTransferAmount()) {
+            return this._remainingMakerTokenAmount;
+        }
+        if (this._signedOrder.makerFee.isZero()) {
+            return BigNumber.min(this._remainingMakerTokenAmount, this._transferrableMakerTokenAmount);
+        } else {
+            return this.calculatePartiallyFillableMakerTokenAmount();
+        }
+    }
+    public computeRemainingTakerFillable(): BigNumber {
+        return this.computeRemainingMakerFillable().times(this._signedOrder.takerTokenAmount)
+                                                   .dividedToIntegerBy(this._signedOrder.makerTokenAmount);
+    }
+    private hasSufficientFundsForFeeAndTransferAmount(): boolean {
+        if (this._isMakerTokenZRX) {
+            const totalZRXTransferAmountRequired = this._remainingMakerTokenAmount.plus(this._remainingMakerFeeAmount);
+            return this._transferrableMakerTokenAmount.gte(totalZRXTransferAmountRequired);
+        } else {
+            const hasSufficientFundsForTransferAmount = this._transferrableMakerTokenAmount.gte(
+                                                            this._remainingMakerTokenAmount);
+            const hasSufficientFundsForFeeAmount = this._transferrableMakerFeeTokenAmount.gte(
+                                                       this._remainingMakerFeeAmount);
+            return (hasSufficientFundsForTransferAmount && hasSufficientFundsForFeeAmount);
+        }
+    }
+
+    private calculatePartiallyFillableMakerTokenAmount(): BigNumber {
+        const orderToFeeRatio = this._signedOrder.makerTokenAmount.dividedToIntegerBy(this._signedOrder.makerFee);
+        // Maximum number of times the Maker can fill the order, given the fees
+        const fillableTimesInFeeTokenUnits = BigNumber.min(this._transferrableMakerFeeTokenAmount,
+                                                           this._remainingMakerFeeAmount);
+        // Maximum number of times the Maker can fill the order, given the Maker Token Balance
```

Comment:
I don't understand those `number of times` comments. I can fill an order for 1 3 times or for 3 1 time.